src/native/ceph_s3select/ceph_s3lect_napi.cpp

```diff
@@ -0,0 +1,428 @@
+        this->SuppressDestruct();
+    }
+
+    /*~CephSelectWorker(){
+        std::cout << "dtor worker" << std::endl;
+        std::cout.flush();
+
+    }*/
+
+	void Execute() override
+	{
+		//DBG1("FS::FSWorker::Execute: " << _desc << DVAL(_uid) << DVAL(_gid) << DVAL(_backend));
+
+        /*std::cout << "exec len" << buffer_len << " is flush " << is_flush << std::endl;
+        std::cout.flush();*/
+        //aws.init_result();
+
+        std::string str1;
+
+        this->select.clear();
+        this->chunk.clear();
+
+        int rc;
+        if (csv == input_format) {
+            if (is_flush) {
+                rc = csv_object->run_s3select_on_stream(this->select, this->buffer, 0, 0);
+            } else {
+                rc = csv_object->run_s3select_on_stream(this->select, this->buffer, this->buffer_len, SIZE_MAX);
+            }
+        } else {//currenty only json
+            if (is_flush) {
+                rc = json_object->run_s3select_on_stream(this->select, this->buffer, 0, 0);
+            } else {
+                rc = json_object->run_s3select_on_stream(this->select, this->buffer, this->buffer_len, SIZE_MAX);
+            }
+        }
+	    if (rc < 0) {
+            throw Napi::Error::New(Env(), XSTR() << "s3select: csv.run_s3select_on_stream failed " << csv_object->get_error_description());
+	    }
+
+        //std::cout << "res.len = " << aws.get_result().length() << " payload line len = " << strlen(PAYLOAD_LINE) << std::endl;
+        if(this->select.length() > 0) {
+            aws.run_s3select(this->select);
+            //std::cout << "res = " << aws.get_result() << std::endl;
+            //this->output2 = aws.get_result();
+
+
+
+            /*std::cout << "chunk = " << this->chunk << std::endl;
+            std::cout << "end ================================================ " << std::endl;*/
+
+            /*std::cout << "aws get result = " << aws.get_result() << std::endl;
+            std::cout << "end ================================================ " << std::endl;*/
+
+            std::cout.flush();
+        }
+
+
+	    /*std::cout << "select res = " << this->output << std::endl;
+        std::cout.flush();*/
+	}
+
+    void OnOK() override
+    {
+        //DBG1("FS::FSWorker::OnOK: output " << output);
+        Napi::Env env = Env();
+        Napi::Value napi_res = Napi::Buffer<char>::/*New*/Copy(env, chunk.data(), chunk.length());
+        this->deferred->Resolve(napi_res);
+    }
+
+    void OnError(Napi::Error const& error) override
+    {
+        Napi::Env env = Env();
+        //DBG1("FS::FSWorker::OnError: " << _desc << " " << DVAL(error.Message()));
+        //ReportWorkerStats(1);
+        auto obj = error.Value();
+        /*if (_errno) {
+            obj.Set("code", Napi::String::New(env, uv_err_name(uv_translate_sys_error(_errno))));
+        }*/
+        //_deferred.Reject(obj);
+    }
+};
+
+class CephS3lectNapi : public Napi::ObjectWrap<CephS3lectNapi> {
```

Comment:
I don't think the class name needs to have Ceph in its name.
The fact that this library is part of ceph is not important, and it has nothing to do with ceph itself.
I would just name this S3SelectNapi.

Reply:
This is a general comment - I don't think we need to name any of our types as CephXXX. This is confusing and makes readers think that we are using ceph codebase for this functionality, but we only use the s3select library which is a completely decoupled lib.